state/migration_import.go

```diff
@@ -68,8 +71,43 @@ func (st *State) Import(model description.Model) (_ *Model, _ *State, err error)
 		// filesystems or storage instances.
 		StorageProviderRegistry: storage.StaticProviderRegistry{},
 	}
-	if creds := model.CloudCredential(); creds != "" {
-		args.CloudCredential = names.NewCloudCredentialTag(creds)
+	if creds := model.CloudCredential(); creds != nil {
+		// Need to add credential or make sure an existing credential
+		// matches.
+		// TODO: there really should be a way to create a cloud credential
+		// tag in the names package.
+		credID := fmt.Sprintf("%s/%s/%s", creds.Cloud(), creds.Owner(), creds.Name())
+		if !names.IsValidCloudCredential(credID) {
+			return nil, nil, errors.Errorf("model credentails id not valid: %q", credID)
+		}
+		credTag := names.NewCloudCredentialTag(credID)
```

Comment:
Really this should take the three values rather than one. (Although not in this change.)